types/models.d.ts

```diff
@@ -245,6 +245,8 @@ declare module 'mongoose' {
      */
     createCollection<T extends mongodb.Document>(options?: mongodb.CreateCollectionOptions & Pick<SchemaOptions, 'expires'>): Promise<mongodb.Collection<T>>;
 
+    createSearchIndex(description: mongodb.SearchIndexDescription): Promise<string>;
```

Comment:
maybe consider also adding tsdoc to the types mirroring the jsdoc (including the note about "atlas only")